scripts/doc-checker.py

```diff
@@ -102,22 +142,96 @@ def pydocstyle_validate(path, add_ignore):
     return result.returncode
 
 
-@click.command()
-@click.argument("paths", nargs=-1, required=True, type=click.Path(exists=True))
-@click.option("--add-ignore", multiple=True, help="Pydocstyle errors")
-@click.option("--use-numpydoc", type=bool, default=True)
-def validate(paths, add_ignore, use_numpydoc):
+def monkeypatching():
+    """Monkeypatch decorators which incorrectly define __doc__ attribute."""
+    import ray
+
+    ray.remote = lambda *args, **kwargs: args[0]
+
+
+def validate(
+    paths: List[pathlib.Path], add_ignore: List[str], use_numpydoc: bool
+) -> int:
+    """
+    Perform pydocstyle and numpydoc checks.
+
+    Parameters
+    ----------
+    paths : List[pathlib.Path]
+        filenames of directories for check
+    add_ignore : List[str]
+        pydocstyle error codes which are not verified
+    use_numpydoc : bool
+        determine if numpydoc checks are not needed
+
+    Returns
+    -------
+    exit_status : int
+    """
     exit_status = 0
     for path in paths:
         if pydocstyle_validate(path, add_ignore):
             exit_status = 1
         if use_numpydoc:
+            monkeypatching()
             if numpydoc_validate(path):
                 exit_status = 1
-    if exit_status:
-        exit(exit_status)
-    print("SUCCESSFUL CHECK")
+    return exit_status
+
+
+def check_args(args: argparse.Namespace):
+    """
+    Check the obtained values for correctness.
+
+    Raises
+    ------
+    ValueError
+        Occurs in case of non-existent files or directories.
+    """
+    for path in args.paths:
+        if not path.exists():
+            raise ValueError(f"{path} is not exist")
```

Comment:
```suggestion
            raise ValueError(f"{path} does not exist")
```
I don't think this check is strictly needed, though, as non-existant files would surface later on... but it's better to fail early